core/templates/dev/head/pages/admin/roles_tab/roles_tab_directive.html

```diff
@@ -0,0 +1,96 @@
+                  <option value="username">By Username</option>
+              </select>
+            </div>
+          </div>
+
+          <div class="form-group" ng-if="view_form_values.method=='role'">
+            <label class="col-md-4 col-lg-4 col-sm-4 control-label pull-left">Select Role</label>
+            <div class="col-md-8 col-lg-8 col-sm-8">
+              <select ng-options="role_name as role_string for (role_name, role_string) in view_role_options" ng-model="view_form_values.role" class="form-control">
+              </select>
+            </div>
+          </div>
+
+          <div class="form-group" ng-if="view_form_values.method=='username'">
+            <label class="col-md-4 col-lg-4 col-sm-4 control-label">Enter Username</label>
+            <div class="col-md-8 col-lg-8 col-sm-8">
+              <input type="text" placeholder="Enter username" ng-model="view_form_values.username" class="form-control" />
+            </div>
+          </div>
+
+          <button type="submit" class="btn btn-success" ng-disabled="view_form_values.method==None || (view_form_values.role==undefined && view_form_values.username==undefined)" value="view role">View Role</button>
+        </form>
+
+        <div ng-show="show_result_roles">
+          <br/>
+          <table class="table text-center">
+            <thead>
+              <tr>
+                <td><strong>Username</strong></td>
+                <td><strong>Role</strong></td>
+              </tr>
+            </thead>
+            <tbody>
+              <tr ng-repeat="(name, role) in result">
+                  <td>{{name}}</td>
+                  <td>{{role}}</td>
+              </tr>
+            </tbody>
+          </table>
+        </div>
+      </md-card>
+
+      <md-card class="oppia-page-card oppia-long-text update-role-form">
+        <form class="form-horizontal" ng-submit="SubmitUpdateRoleForm(update_form_values)">
+          <legend class="text-center">Update Role</legend>            
+          <div class="form-group">
+              <label class="col-md-4 col-lg-4 col-sm-4 control-label">Enter Username</label>
+              <div class="col-md-8 col-lg-8 col-sm-8">
+                <input type="text" placeholder="Enter username" ng-model="update_form_values.username" class="form-control" />
+              </div>
+          </div>
+
+          <div class="form-group">
+            <label class="col-md-4 col-lg-4 col-sm-4 control-label pull-left">Select Role</label>
+            <div class="col-md-8 col-lg-8 col-sm-8">
+              <select ng-options="role_name as role_string for (role_name, role_string) in update_role_options" ng-model="update_form_values.newrole" class="form-control">
+              </select>
+            </div>
+          </div>
+  
+          <button type="submit" class="btn btn-success" ng-disabled="update_form_values.newrole==undefined || update_form_values.username==undefined" value="update role">Update Role</button>
```

Comment:
newrole --> newRole

Also, use camelCase and not snake_case for frontend stuff.